project/src/components/sort/sort.tsx

```diff
@@ -1,18 +1,47 @@
-function Sort(): JSX.Element {
+import { useState } from 'react';
+import cn from 'classnames';
+import { SortType } from '../../const';
+import { useAppDispatch } from '../../hooks/use-app-dispatch/use-app-dispatch';
+import { changeSortType } from '../../store/action';
+
+type SortProps = {
+  currentSortType: SortType;
+}
+
+function Sort({currentSortType}: SortProps): JSX.Element {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const dispatch = useAppDispatch();
+
   return (
     <form className="places__sorting" action="#" method="get">
       <span className="places__sorting-caption">Sort by</span>
-      <span className="places__sorting-type" tabIndex={0}>
-        Popular
+      <span
+        className="places__sorting-type"
+        tabIndex={0}
+        onClick={() => setIsOpen(!isOpen)}
+      >
+        {currentSortType}
         <svg className="places__sorting-arrow" width="7" height="4">
           <use xlinkHref="#icon-arrow-select"></use>
         </svg>
       </span>
-      <ul className="places__options places__options--custom">
-        <li className="places__option places__option--active" tabIndex={0}>Popular</li>
-        <li className="places__option" tabIndex={0}>Price: low to high</li>
-        <li className="places__option" tabIndex={0}>Price: high to low</li>
-        <li className="places__option" tabIndex={0}>Top rated first</li>
+      <ul className={cn('places__options places__options--custom', isOpen && 'places__options--opened')}>
+        {Object.values(SortType).map((sortType) => (
+          <li
+            className={cn('places__option', sortType === currentSortType && 'places__option--active')}
```

Comment:
Если не булев параметр предпочитаю форматировать так
```suggestion
            className={cn('places__option', { 'places__option--active': sortType === currentSortType })}
```